flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/stream/StreamExecWatermarkAssigner.scala

```diff
@@ -126,7 +125,10 @@ class StreamExecWatermarkAssigner(
       // 1. redundant watermark definition in DDL
```

Comment:
This is kind of hack IMO, should revisit this in future.

Reply:
Agree. The previous MiniBatchIntervalTrait design doesn't match the new DDL watermark any more. Will revisit and refactor this in [FLINK-14622](https://issues.apache.org/jira/browse/FLINK-14622).